loginserver/world_server.cpp

```diff
@@ -1348,3 +1354,36 @@ void WorldServer::OnKeepAlive(EQ::Timer *t)
 	ServerPacket pack(ServerOP_KeepAlive, 0);
 	m_connection->SendPacket(&pack);
 }
+
+void WorldServer::FormatWorldServerName(char *name, int8 server_list_type)
+{
+	std::string server_long_name = name;
+	server_long_name = trim(server_long_name);
+
+	bool name_set_to_bottom = false;
+	if (server.options.IsWorldDevTestServersListBottom() && server_list_type == LS::ServerType::Standard) {
+		std::string s = str_tolower(server_long_name);
+		if (s.find("dev") != std::string::npos) {
+			server_long_name   = fmt::format("|D| {}", server_long_name);
+			name_set_to_bottom = true;
+		}
+		else if (s.find("test") != std::string::npos) {
```

Comment:
Should this look for whole words instead? It could catch false positives if a server name has `test` as part of another word. `The greatest server` for example

Reply:
It could, but then we'd be missing things like `(Test)` or `[Test]` and I don't want to overly science this. You could strip special characters before doing the match but then someone else prepends a letter to bypass it. You're going to have issues one way or another